internal/cmd/plugin/plugin_test.go

```diff
@@ -17,23 +17,74 @@ limitations under the License.
 package plugin
 
 import (
+	"context"
+	"fmt"
+	"io"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"net/http"
 	k8client "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"strings"
+
+	"k8s.io/client-go/rest"
 
 	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
 	"github.com/cloudnative-pg/cloudnative-pg/internal/scheme"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/versions"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
+type fakeRoundTripper struct {
```

Comment:
Not sure if these should be extracted to a shared/common location? This is my first time writing Go tests; not sure if I missed an established way to do this within the codebase.